### ISSUE TITLE
Unhandled Exception on Telemetry.Flush() when Statsd server is not up

### ISSUE THREAD

**Issue by PSanetra** [DataDog/dogstatsd-csharp-client#118]
When Telemetry is enabled and the Statsd server is not running, the following Exception is unhandled:

```
Unhandled exception. System.Net.Sockets.SocketException (111): Connection refused
   at StatsdClient.StatsSender.Send(Byte[] buffer, Int32 length)
   at StatsdClient.Telemetry.SendMetric(String metricName, Int32 value)
   at StatsdClient.Telemetry.Flush()
   at StatsdClient.Telemetry.<.ctor>b__13_0(Object _)
   at System.Threading.TimerQueueTimer.<>c.<.cctor>b__23_0(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.TimerQueueTimer.CallCallback(Boolean isThreadPool)
   at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool)
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.TimerQueue.AppDomainTimerCallback(Int32 id)
```

**Comment by ogaca-dd** [comment]
@PSanetra ,

Thank you for reporting this issue.
https://github.com/DataDog/dogstatsd-csharp-client/pull/120 should fix it.

**Comment by ogaca-dd** [comment]
@PSanetra ,

It should be fixed by [DogStatsD-CSharp-Client 5.0.1](https://www.nuget.org/packages/DogStatsD-CSharp-Client).
Let us know if you still have the issue.

**Comment by PSanetra** [comment]
Looks good with 5.0.1 :thumbsup:
